src/js/operations/Extract.js

```diff
@@ -293,5 +293,66 @@ var Extract = {
         output += Extract.run_dates(input, []);
         return output;
     },
-    
+
+    /**
+     * @constant
+     * @default
+     */
+    XPATH_INITIAL: "",
+
+    /**
+     * @constant
+     * @default
+     */
+    XPATH_DELIMITER: "\\n",
+
+    /**
+     * Extract information (from an xml document) with an XPath query
+     *
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run_xpath:function(input, args) {
+        var query = args[0];
+        var delimiter = args[1];
+
+        try {
+            var xml = $.parseXML(input);
+        } catch (err) {
+            return "Invalid input XML.";
+        }
+
+        try {
+            var result = $.xpath(xml, query);
+        } catch (err) {
+            return "Invalid XPath. Details:\n" + err.message;
+        }
+
+        var serializer = new XMLSerializer();
+        var output = "";
+        for (var i = 0; i < result.length; i++) {
```

Comment:
This can be done much more concisely and you can avoid a for-loop:

```js
function nodeToString(node) {
  const { nodeType, value, wholeText, data} = node;
  switch (nodeType) {
    case Node.ELEMENT_NODE: return serializer.serializeToString(node);
    case Node.ATTRIBUTE_NODE: return value;
    case Node.COMMENT_NODE: return data;
    default:  throw new Error(`Unknown Node Type: ${nodeType}`);
  }
}

return result
  .map(nodeToString)
  .join(delimiter);
```